semantic/src/main/kotlin/tools/samt/semantic/SemanticModel.kt

```diff
@@ -0,0 +1,597 @@
+                            }
+                        }
+                    }
+                }
+                when (import) {
+                    is TypeImportNode -> {
+                        // Just import one type
+                        val type = global.resolveType(import.name)
+                        if (type != null) {
+                            val name = if (import.alias != null) {
+                                import.alias!!.name
+                            } else {
+                                import.name.components.last().name
+                            }
+
+                            addImportedType(name, type)
+                        } else {
+                            controller.createContext(file.sourceFile).error {
+                                message("Import '${import.name.name}' could not be resolved")
+                                highlight("unresolved import", import.location)
+                            }
+                        }
+                    }
+
+                    is WildcardImportNode -> {
+                        // Import all types from the package
+                        val type = global.resolveType(import.name)
+                        if (type != null) {
+                            if (type is PackageType) {
+                                type.sourcePackage.types.forEach { (name, type) ->
+                                    addImportedType(name, type)
+                                }
+                            } else {
+                                controller.createContext(file.sourceFile).error {
+                                    message("Import '${import.name.name}.*' must point to a package and not a type")
+                                    highlight(
+                                        "illegal wildcard import", import.location, suggestChange = "import ${
+                                            import.name.components.dropLast(1).joinToString(".") { it.name }
+                                        }.*")
+                                }
+                            }
+                        } else {
+                            controller.createContext(file.sourceFile).error {
+                                message("Import '${import.name.name}.*' could not be resolved")
+                                highlight("unresolved import", import.location)
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Add all top-level packages to scope
+            val topLevelPackages = global.subPackages.map { PackageType(it) }
+            topLevelPackages.forEach { putIfAbsent(it.packageName, it) }
+
+            // Add built-in types
+            fun addBuiltIn(name: String, type: Type) {
+                putIfAbsent(name, type)?.let { existingType ->
+                    controller.createContext(file.sourceFile).warn {
```

Comment:
Maybe make this an error? What's the reason for being able to override builtin types?

Reply:
changed to error